openassessment/xblock/staff_assessment_mixin.py

```diff
@@ -0,0 +1,102 @@
+"""
+A mixin for staff grading.
+"""
+from openassessment.assessment.api import staff as staff_api
+from staff_area_mixin import require_course_staff
+from xblock.core import XBlock
+
+from .data_conversion import create_rubric_dict
+from .resolve_dates import DISTANT_FUTURE
+from .data_conversion import clean_criterion_feedback, create_submission_dict
+
+
+class StaffAssessmentMixin(object):
+    """
+    This mixin is for all staff-assessment related endpoints.
+    """
+
+    @XBlock.json_handler
+    @require_course_staff("STUDENT_INFO")
+    def staff_assess(self, data, suffix=''):
+        """
+        Create a staff assessment from a staff submission.
+        """
+        if 'options_selected' not in data:
+            return {'success': False, 'msg': self._(u"Missing options_selected key in request")}
+
+        if 'overall_feedback' not in data:
+            return {'success': False, 'msg': self._('Must provide overall feedback in the assessment')}
+
+        if 'criterion_feedback' not in data:
+            return {'success': False, 'msg': self._('Must provide feedback for criteria in the assessment')}
+
+        if self.submission_uuid is None:
+            return {'success': False, 'msg': self._(u"You must submit a response before you can perform a self-assessment.")}
```

Comment:
I believe this check is needed, but you're right that I should probably change it to take an arbitrary submission id.